pages/in/widget/iscn-ar/index.vue

```diff
@@ -0,0 +1,746 @@
+        error({ statusCode: 400, message: 'RECEIPIENT_HAS_NO_WALLET' });
+      }
+      const hasOpener = opener && opener !== '0';
+
+      return {
+        toIds,
+        amounts,
+        toUsers,
+        redirectUri,
+        remarks,
+        opener: hasOpener,
+        title,
+        tags,
+        url,
+        fingerprints,
+        license,
+        publisher,
+      };
+    }).catch((e) => {
+      console.error(e);
+      error({ statusCode: 404, message: e.message });
+    });
+  },
+  head() {
+    return {
+      title: 'LIKE pay',
+    };
+  },
+  computed: {
+    ...mapGetters([
+      'getUserInfo',
+      'getIsSignFinishedState',
+      'getIsTxFailed',
+    ]),
+    redirectOrigin() {
+      const url = new URL(this.redirectUri, true);
+      return url.origin;
+    },
+    windowOpener() {
+      if (!window) return null;
+      return window.opener;
+    },
+    isChainUpgrading() {
+      return IS_CHAIN_UPGRADING;
+    },
+    sumOfToAmount() {
+      if (!this.amounts) return '0';
+      const amount = this.amounts.reduce(
+        (acc, a) => acc.plus(a),
+        new BigNumber(0),
+      );
+      return amount.toFixed();
+    },
+    actualSendAmount() {
+      const amount = new BigNumber(this.sumOfToAmount);
+      return amount.toFixed();
+    },
+    totalAmount() {
+      let amount = new BigNumber(this.actualSendAmount);
+      if (this.gasFee) amount = amount.plus(this.gasFee);
+      return amount.toFixed();
+    },
+    likePayMetadata() {
+      const {
+        toIds,
+        amounts,
+        redirectUri,
+        remarks,
+      } = this;
+      return {
+        likePay: {
+          toIds,
+          amounts,
+          redirectUri,
+          remarks,
+        },
+      };
+    },
+  },
+  async mounted() {
+    const {
+      fingerprints,
+      title,
+      tags,
+      type,
+      license,
+      url,
+    } = this;
+    const { cosmosWallet } = this.getUserInfo;
+    if (!this.isChainUpgrading) {
+      this.calculateGasFee();
+    }
+    if (this.opener && !window.opener) {
+      this.$nuxt.error({ statusCode: 400, message: 'Cannot access window opener' });
+    }
+    this.ISCNTotalFee = await this.calculateISCNTxTotalFee({
+      userId: this.getUserId,
+      displayName: this.getUserInfo.displayName,
+      cosmosWallet,
+      fingerprints,
+      name: title,
+      tags,
+      type,
+      license,
+      url,
+    });
+  },
+  methods: {
+    ...mapActions([
+      'sendCosmosPayment',
+      'fetchCurrentCosmosWallet',
+      'prepareCosmosTxSigner',
+      'setDefaultCosmosWalletSource',
+      'calculateISCNTxTotalFee',
+      'sendISCNSignature',
+    ]),
+    async calculateGasFee() {
+      const { feeAmount } = await calculateCosmosGas(this.toUsers);
+      this.gasFee = BigNumber(feeAmount[0].amount).dividedBy(1e9).toFixed();
+      return this.gasFee;
+    },
+    async onArweaveUploadSuccessStartRegisterISCNMessage(event) {
```

Comment:
Please rename this `onArweaveUploadSuccessStartRegisterISCNMessage` to `onStartRegisterISCNMessage`